File: selfstudy/golang/stars_avg.go
/*
task 284:
Stars - Avg
You need to compute the average of star luminosities
*/

package main

import (
	"fmt"
	"reflect"
)

func AverageStars(brightness []int) float64 {
	if len(brightness) == 0 { return 0 }
	sum := 0
	for _, v := range brightness {
		sum += v
	}
	return float64(sum)/float64(len(brightness))
}

func main() {
	tests := []struct {
		input []int
		expected float64
	}{
		{[]int{10, 20, 30}, AverageStars([]int{10, 20, 30})},
		{[]int{50, 60, 70}, AverageStars([]int{50, 60, 70})},
		{[]int{1, 2, 3}, AverageStars([]int{1, 2, 3})},
		{[]int{100, 200}, AverageStars([]int{100, 200})},
		{[]int{}, AverageStars([]int{})},
	}
	
	for idx, t := range tests {
		res := AverageStars(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done

File: selfstudy/js/stars_avg.js
/*
task 84:
Stars - Avg
You need to compute the average of star luminosities
*/

function average_stars(brightness) {
  return brightness.length ? brightness.reduce((a, b) => a + b, 0) / brightness.length : 0;
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: average_stars([10, 20, 30]) },
  { input: [50, 60, 70], expected: average_stars([50, 60, 70]) },
  { input: [1, 2, 3], expected: average_stars([1, 2, 3]) },
  { input: [100, 200], expected: average_stars([100, 200]) },
  { input: [], expected: average_stars([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('average_stars', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(average_stars(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, average_stars(t.input) === t.expected || JSON.stringify(average_stars(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// works i think
